File: popup.js
document.getElementById('scanBtn').addEventListener('click', scanUrl);

async function scanUrl() {
  const url = document.getElementById('urlInput').value;
  const resultDiv = document.getElementById('result');

  if (!url) {
    alert('Please enter a URL');
    return;
  }

  resultDiv.textContent = 'Scanning...';
  resultDiv.className = '';
  resultDiv.style.display = 'block';

  try {
    const response = await fetch('http://localhost:5000/api/check-url', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ url })
    });

    const data = await response.json();

    if (data.suspicious) {
      resultDiv.className = 'danger';
      resultDiv.innerHTML = `
        <strong>⚠️ SUSPICIOUS!</strong><br>
        Score: ${data.score}/100<br>
        ${data.reasons.join('<br>')}
      `;
    } else {
      resultDiv.className = 'safe';
      resultDiv.innerHTML = '<strong>✅ Safe</strong><br>No threats detected';
    }
  } catch (error) {
    resultDiv.className = 'danger';
    resultDiv.textContent = 'Error: Could not connect to API';
  }
}

File: extension/popup.py
document.addEventListener('DOMContentLoaded', () => {
  const urlInput = document.getElementById('url');
  const modeSelect = document.getElementById('mode');
  const scanBtn = document.getElementById('scan');
  const resultDiv = document.getElementById('result');

  // Load saved mode
  chrome.storage.local.get({ mode: 'heuristic' }, (items) => {
    modeSelect.value = items.mode || 'heuristic';
  });

  modeSelect.addEventListener('change', () => {
    const mode = modeSelect.value;
    chrome.storage.local.set({ mode });
    resultDiv.textContent = `Mode set to ${mode}.`;
  });

  scanBtn.addEventListener('click', async () => {
    const url = urlInput.value.trim();
    const mode = modeSelect.value;

    if (!url) {
      resultDiv.textContent = 'Enter a URL to scan.';
      return;
    }

    resultDiv.textContent = 'Scanning...';

    chrome.runtime.sendMessage({ action: 'popup_scan', url, mode }, (resp) => {
      if (!resp) {
        resultDiv.textContent = 'No response (backend may be unreachable).';
        return;
      }
      // Normalize
      let verdict = resp.verdict || (resp.result && resp.result.verdict) || 'unknown';
      let score = resp.score || (resp.result && resp.result.score) || (resp.ti && (resp.ti.stats && (resp.ti.stats.malicious || 0))) || 0;
      let reasons = resp.reasons || (resp.result && resp.result.reasons) || [];
      resultDiv.innerHTML = `<strong>Verdict:</strong> ${verdict.toUpperCase()}<br><strong>Score:</strong> ${score}<br><strong>Reasons:</strong> ${reasons.join(', ') || 'none'}`;
    });
  });
});
